// $custom/schema/index.js
const config 		= require('$custom/config');
const mongoose 	= require('mongoose');

mongoose.connect(config.mongoose);

const Schema = {
	Commits: mongoose.Schema({
	  commit_id: String,
	  repository_id: Number,
	  author: String,
	  email: String,
		subject: String,
		comment: String,
		logs: Boolean,
	  since: Date
	}),
	Repository: mongoose.Schema({
	  repository_id: Number,
	  collection_name: String,
	  repository_name: String,
	  title: String,
	  description: String,
	  master: String,
	  branch: [String],
	  files: [{
	  	filename: String, 
	  	ext: String, 
	  	size: Boolean, 
	  	since: Date, 
	  	comment: String
	  }]
	}),
	Clients: mongoose.Schema({
	  user_id: Number,
	  fullname: String,
	}),
	Users: mongoose.Schema({
	 any: {} 
	})
}

module.exports = {
	Commit: mongoose.model('logs_commits', Schema.Commits),
	Client: mongoose.model('logs_clients', Schema.Clients),
	Repository: mongoose.model('cache_repository', Schema.Repository),
	User: mongoose.model('users', Schema.Users),
}
