// $custom/config/index.js
var getConfig = process.env.METEOR_CONFIG || 'default';

module.exports = ({
	'dev': {
		arg: getConfig,
		platfrom: 'WINDOWS',
		domain: '//localhost:8100/',
		socket: 'http://localhost:8200',
		git: 8100,
		api: 8200,
		mongoose: 'mongodb://touno-k.com/debugerr-sentinel',
		mysql: { usernamne:'', password:'', hostname:'', port: 3306, dbname:'' },
		source: './tmp/source',
		fork: './tmp/package',
		content: './tmp/content',
		core: 'C:/Program Files/Git/mingw64/libexec/git-core',
		lfs: 'C:/Program Files/Git LFS'
	},
	'local': {
		arg: getConfig,
		platfrom: 'WINDOWS',
		domain: '//localhost:8100/',
		socket: 'http://192.168.10.6:8200',
		git: 8100,
		api: 8200,
		mongoose: 'mongodb://touno-k.com/debugerr-sentinel',
		mysql: { usernamne:'root', password:'ar00t', hostname:'db.touno-k.com', port: 3306, dbname:'debugerr' },
		source: './tmp/source/',
		fork: './tmp/package',
		content: './tmp/content',
		core: 'C:/Program Files/Git/mingw64/libexec/git-core',
		lfs: 'C:/Program Files/Git LFS'
	},
	'default': {
		arg: getConfig,
		platfrom: 'LINUX',
		domain: '//dev.ns.co.th:810/',
		socket: 'http://local-sentinel:811',
		git: 810,
		api: 811,
		mongoose: 'mongodb://local-mongodb:27017/debugerr-sentinel',
		mysql: { usernamne:'', password:'', hostname:'', port: 3306, dbname:'' },
		source: '/data/source',
		fork: './data/package',
		content: './data/content',
		core: '/usr/bin',
		lfs: '/usr/bin'
	}
})[getConfig]
